splice/web/api/tile.py

```
@@ -0,0 +1,125 @@
+                                   help='Attribution URL', location='json')
+        self.reqparse.add_argument('attribution_period', type=int,
+                                   help='Attribution period', location='json')
+        self.reqparse.add_argument('image_uri', type=str, required=True,
+                                   help='base64 encoded roll over image', location='json')
+        self.reqparse.add_argument('enhanced_image_uri', type=str, required=True,
+                                   help='base64 encoded static image', location='json')
+        self.reqparse.add_argument('bg_color', type=str, default="",
+                                   help='Background color', location='json')
+        self.reqparse.add_argument('explanation', type=unicode, default="",
+                                   help='Contextual explanation', location='json')
+        super(TileListAPI, self).__init__()
+
+    def get(self, adgroup_id):
+        tiles = get_tiles_by_adgroup_id(adgroup_id)
+        if len(tiles) == 0:
+            return {"message": "No tiles found"}, 404
+        else:
+            return {"message": marshal(tiles, tile_fields)}
+
+    def post(self, adgroup_id):
+        """ HTTP end point to create new tile. Note the initial status of a new
+        tile is always set as 'unapproved'
+        """
+        args = self.reqparse.parse_args()
+        # the status of new tiles are alway set as unapproved
+        args["status"] = "unapproved"
+
+        try:
+            # validate background color, url, and image fields again
+            jsonschema.validate(args, API_TILE_SCHEMA)
+            with session_scope() as session:
+                new = insert_tile(session, args)
+        except jsonschema.exceptions.ValidationError as e:
+            return {"message": e.message}, 400
+        except InvalidRequestError as e:
+            return {"message": e.message}, 400
+        except Exception as e:
```

Yes, this is not sufficient. It swallows all the exceptions, and sometime even returns some confusing message back to the client. A better exception handling and logging are definitely needed here.